dspace-api/src/main/java/org/dspace/importer/external/ris/service/RisImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.ris.service;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Resource;
+
+import org.dspace.importer.external.exception.FileSourceException;
+import org.dspace.importer.external.service.components.AbstractPlainMetadataSource;
+import org.dspace.importer.external.service.components.dto.PlainMetadataKeyValueItem;
+import org.dspace.importer.external.service.components.dto.PlainMetadataSourceDto;
+
+/**
+ * Implements a metadata importer for RIS files
+ * Implementations insprider by BTE DataLoader {@link https://github.com/EKT/Biblio-Transformation-Engine/blob/master/bte-io/src/main/java/gr/ekt/bteio/loaders/RISDataLoader.java}
+ * 
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ */
+public class RisImportMetadataSourceServiceImpl extends AbstractPlainMetadataSource {
+
+    @Override
+    public String getImportSource() {
+        return "RISMetadataSource";
+    }
+
+    protected List<PlainMetadataSourceDto> readData(InputStream inputStream) throws FileSourceException {
+        return aggreageteData(inputStream);
+    }
+
+    private List<PlainMetadataSourceDto> aggreageteData(InputStream inputStream) throws FileSourceException {
+        List<PlainMetadataSourceDto> metadata = new ArrayList<>();
+        List<PlainMetadataKeyValueItem> notAggregatedItems = notAggregatedData(inputStream);
+        List<PlainMetadataKeyValueItem> aggregatedTmpList = null;
+        Iterator<PlainMetadataKeyValueItem> itr = notAggregatedItems.iterator();
+        while (itr.hasNext()) {
+            PlainMetadataKeyValueItem item = itr.next();
+            if ("TY".equals(item.getKey())) {
+                if (aggregatedTmpList != null) {
+                    PlainMetadataSourceDto dto = new PlainMetadataSourceDto();
+                    dto.setMetadata(new ArrayList<>(aggregatedTmpList));
+                    metadata.add(dto);
+                }
+                aggregatedTmpList = new ArrayList<>();
+                aggregatedTmpList.add(item);
+            } else {
+                if (aggregatedTmpList != null) {
+                    aggregatedTmpList.add(item);
+                    // save last iteration metadata
+                    if (!itr.hasNext()) {
+                        PlainMetadataSourceDto dto = new PlainMetadataSourceDto();
+                        dto.setMetadata(new ArrayList<>(aggregatedTmpList));
+                        metadata.add(dto);
+                    }
+                }
+            }
+        }
+        return metadata;
+    }
+
+    private List<PlainMetadataKeyValueItem> notAggregatedData(InputStream inputStream) throws FileSourceException {
```

Comment:
Same here, this is an important method & I'd like to see JavaDocs to describe it (as it will make it easier to maintain or debug in the future)